### ISSUE TITLE
Debugging option (Attempting to run on OpenBSD)

### ISSUE THREAD

**Issue by OpenBSDGuy** [pystardust/ytfzf#544]
### Before Making This Post
* [x] I have taken a look at both man pages, `ytfzf(1)` and `ytfzf(5)`

### What do you need help with?
Hi,

This project is awesome. I'm trying to make it run on OpenBSD. But when I'm trying to play or download a video, it quits without any error. I'm wondering is there an easy way to enable some logging besides either removing `/dev/null` pipes or intercepting `/dev/null`?

To demonstrate better, here is the screenshot which I captured after it quits
![2022-06-02-171228_724x295_scrot](https://user-images.githubusercontent.com/96616294/171661979-9011c132-3bc8-42e4-ab2e-bc81f64678b0.png)

**Comment by Euro20179** [comment]
Very interesting.

First, what is `sh` symlinked to (to check, `ls -l $(which sh)`), it could be part of the problem, but i find it unlikely.

Also before running the somewhat sketchy debugging method, try using `ytfzf -a stallman` just to see if it is in fact scraping videos but breaking for some reason.

If it doesn't work, or if you want to try it anyway:

`sh -x ytfzf -D stallman 2> log-file`

Problem with this is that this absolutely destroys fzf, so make sure you have `dmenu` installed so that -D works.

**Comment by OpenBSDGuy** [comment]
Thanks for the reply.

The output of the `ls -l $(which sh)` is

```bash
-r-xr-xr-x  3 root  bin  617576 Apr  8 18:23 /bin/sh
```

When I ran `ytfzf -a stallman`, the video started to play after a few seconds.

This is the log I've captured [log-file.log](https://github.com/pystardust/ytfzf/files/8839580/log-file.log).

I read [`sh` man page](https://man.openbsd.org/sh.1) on OpenBSD and something caught my eyes:

```
This version of sh is actually ksh in disguise.  As such, it also supports the features described in ksh(1).
```

Since I also have `bash` installed on my computer,  I changed the `ytfzf` first line from,

```bash
#!/usr/bin/env sh
```
to
```bash
#!/usr/bin/env bash
```

And it starts working without any issue.

But out of curiosity I went and remove first `2 > /dev/null` and got error output like this

```bash
/ytfzf[2118]: |21837: unexpected `|'
./ytfzf[2118]: |363177: unexpected `|'
./ytfzf[2118]: |38373: unexpected `|'
./ytfzf[2118]: |84376: unexpected `|'
./ytfzf[2118]: |1351098: unexpected `|'
./ytfzf[2118]: |63499: unexpected `|'
./ytfzf[2118]: |17327: unexpected `|'
./ytfzf[2118]: |133467: unexpected `|'
./ytfzf[2118]: |3151: unexpected `|'
./ytfzf[2118]: |25065: unexpected `|'
./ytfzf[2118]: |22320: unexpected `|'
./ytfzf[2118]: |321030: unexpected `|'
./ytfzf[2118]: |2197: unexpected `|'
./ytfzf[2118]: |6991: unexpected `|'
./ytfzf[2118]: |1950: unexpected `|'
./ytfzf[2118]: |2279: unexpected `|'
./ytfzf[2118]: |8063: unexpected `|'
./ytfzf[2118]: |10238: unexpected `|'
./ytfzf[2118]: |897: unexpected `|'
./ytfzf[2118]: |33374: unexpected `|'
```
`2118` is basically pointing to the `main` line. My line number probably differs since I added bunch of logs here and there. However, I won't get such logs when running it with `bash` explicitly. I added some logs and narrowed down the issue to `open_url_handler` function. On OpenBSD `sh` (basically `ksh`) somewhat this condition `[ -z "$*" ] && return 0` is true and jumps out of the ytfzf without any error. 

I'm guessing something prior to that fails that causes the condition to be true. Since OpenBSD uses its own `ksh` (fork of `pdksh` since many many years ago) I am not sure how feasible would be to resolve the issue.

**Comment by Euro20179** [comment]
I had a feeling it had to do with ksh because bsd, but when i tried it with ksh it worked! Ill do a bit more digging If you want, feel free to make a wrapper script around ytfzf that uses bash.

Thanks for the help!

**Comment by OpenBSDGuy** [comment]
Thanks a lot. I've tested it and it's working now, meaning I can play the video. Still in the log I get some `unexpected '|'` messages. I will do more testing on other functionalities and update you.

Any chance of this fix goes to master branch? Or it breaks compatibility with other shells?

**Comment by Euro20179** [comment]
It shouldn't break other shells as all the quotes do in any shell is say this is a literal. This will get to the master branch in 2.3.1 I forget if I've already pushed it to the development branch, if I haven't I will.

**Comment by Euro20179** [comment]
This fix is now in 2.4
